src/main/java/com/iexec/core/chain/DealWatcherService.java

```diff
@@ -95,11 +111,10 @@ private void onDealEvent(DealEvent dealEvent) {
         BigInteger dealBlock = dealEvent.getBlockNumber();
         log.info("Received deal [dealId:{}, block:{}]", dealId,
                 dealBlock);
-        if (dealBlock == null || dealBlock.equals(BigInteger.ZERO)){
+        if (dealBlock.equals(BigInteger.ZERO)) {
```

Comment:
Why can't it be null?

Reply:
This is directly related to commit 77b6300.
The `BigInteger.ZERO` value is enforced instead of null.
This allowed to remove the call to `setBlockNumber` and to work with immutable `DealEvent` instances.

Reply:
I see, thank you!